File: problems/leetcode/575.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    int distributeCandies(vector<int>& candies) {
        unordered_map<int, int> hash;
        int count = 0;
        for (auto &i : candies) {
            if (hash.find(i) == hash.end()) {
                count++;
                hash[i] = 1;
            }
        }
        return count > candies.size() / 2 ? (candies.size() / 2) : count;
    }
};

File: problems/leetcode/575.py
class Solution:
    def distributeCandies(self, candies: List[int]) -> int:
        hash = {}
        for candi in candies:
            hash[candi] = 1;
        if len(hash.keys()) > len(candies) // 2:
            return len(candies) // 2
        else:
            return len(hash.keys())